drivers/flash/flash_sam_ospic.c

```diff
@@ -0,0 +1,389 @@
+/*
+ * Copyright (C) 2025 Microchip Technology Inc. and its subsidiaries
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+
+#define DT_DRV_COMPAT microchip_sama7g5_qspi
+
+#include <errno.h>
+#include <string.h>
+#include <zephyr/drivers/flash.h>
+#include <zephyr/drivers/dma.h>
+#include <zephyr/cache.h>
+#include "flash_sam_qspi.h"
+
+/* Use PIO for small transfers. */
+#define QSPI_DMA_MIN_BYTES	16
+
+struct sam_qspi_mode {
+	uint8_t cmd_buswidth;
+	uint8_t addr_buswidth;
+	uint8_t data_buswidth;
+	uint32_t config;
+};
+
+static const struct sam_qspi_mode sam_qspi_sama7g5_modes[] = {
+	{ 1, 1, 1, QSPI_IFR_WIDTH_SINGLE_BIT_SPI },
+	{ 1, 1, 2, QSPI_IFR_WIDTH_DUAL_OUTPUT },
+	{ 1, 1, 4, QSPI_IFR_WIDTH_QUAD_OUTPUT },
+	{ 1, 2, 2, QSPI_IFR_WIDTH_DUAL_IO },
+	{ 1, 4, 4, QSPI_IFR_WIDTH_QUAD_IO },
+	{ 2, 2, 2, QSPI_IFR_WIDTH_DUAL_CMD },
+	{ 4, 4, 4, QSPI_IFR_WIDTH_QUAD_CMD },
+	{ 1, 1, 8, QSPI_IFR_WIDTH_OCT_OUTPUT },
+	{ 1, 8, 8, QSPI_IFR_WIDTH_OCT_IO },
+	{ 8, 8, 8, QSPI_IFR_WIDTH_OCT_CMD },
+};
+
+static inline bool sam_qspi_is_compatible(const struct qspi_mem_op *op,
+					    const struct sam_qspi_mode *mode)
+{
+	if (mode->cmd_buswidth != spi_flash_protocol_get_inst_nbits(op->proto)) {
+		return false;
+	}
+	if (mode->addr_buswidth != spi_flash_protocol_get_addr_nbits(op->proto)) {
+		return false;
+	}
+	if (mode->data_buswidth != spi_flash_protocol_get_data_nbits(op->proto)) {
+		return false;
+	}
+	return true;
+}
+
+static int qspi_reg_sync(qspi_registers_t *qspi)
+{
+	uint32_t timeout = 300;
```

Comment:
Should we create a define with the timeout ?
BTW, why some parts use 100ms and here is 300ms ?